src/main/java/com/urbanairship/datacube/dbharnesses/HBaseDbHarness.java

```diff
@@ -268,6 +297,44 @@ private long increment(byte[] rowKey, T op) throws IOException {
         return WithHTable.increment(pool, tableName, rowKey, cf, QUALIFIER, amount);
     }
 
+    /**
+     * @param writes map from rowkey to the operation (which had better be bytes compatible with long)
+     *
+     * @return A map from bytes to the amounts successfully added to the database.
+     *
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    public Map<BoxedByteArray, byte[]> increment(Map<BoxedByteArray, T> writes) throws IOException, InterruptedException {
+        ImmutableMap.Builder<BoxedByteArray, byte[]> successes = ImmutableMap.builder();
+
+        List<Row> increments = new ArrayList<>();
+
+        List<Map.Entry<BoxedByteArray, byte[]>> entries = writes.entrySet().stream()
+                .map(e -> new AbstractMap.SimpleEntry<BoxedByteArray, byte[]>(e.getKey(), e.getValue().serialize()))
```

Comment:
nit: `Maps.immutableEntry()` would be a bit cleaner here.

Reply:
👍. although frankly that whole pre-translation is just generating some extra garbage. I moved it into the for loop.